ui/components/content-renderers/footnote-citation.js

```diff
@@ -1,45 +1,37 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
 
 import { renderContent } from '../../util/render-node';
+import { openFootnote } from '../../store/actions';
 
 import Footnote from '../node-renderers/footnote';
 import Link from '../link';
 
-const propTypes = {
-  content: PropTypes.shape({
-    footnote_node: PropTypes.shape({
-      identifier: PropTypes.string.isRequired,
-    }).isRequired,
-    text: PropTypes.string.isRequired,
-  }).isRequired,
-};
-
-export default class FootnoteCitation extends React.Component {
+export class FootnoteCitation extends React.Component {
   constructor(props) {
     super(props);
 
-    this.state = {
-      expanded: false,
-    };
     this.shrinkFootnote = this.shrinkFootnote.bind(this);
     this.handleCitationClick = this.handleCitationClick.bind(this);
+    this.footnoteIdentifier = props.content.footnote_node.identifier;
   }
 
   handleCitationClick(e) {
     e.preventDefault();
-    this.setState({ expanded: !this.state.expanded });
+    this.props.openFootnote(this.footnoteIdentifier);
```

Comment:
Does this (along with the corresponding logic in the reducer) technically change the behavior of the citation link from _toggling_ the footnote to specifically ensuring that it's open, but not closing it if it's already visible?

Reply:
Excellent catch; I did _not_ intend to change that logic. Will update.